src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -843,6 +843,9 @@ def list_roles(self) -> List[Dict[str, Any]]:
             logger.warning('Error listing roles with code %d: %s', code, err)
             return []
 
+        for role in roles:
+            if not 'PermissionPolicies' in role:
+                role['PermissionPolicies'] = []
```

Comment:
Just a question , thinking on this one.
This will add permission policies to the roles when not present, that means the "PermissionPolicies" field will always be visible on UI , is that okay for UI ?

Reply:
the default behaviour in the details tab is that even if something is empty, its good to show it to the user rather than hide it. In this case, this is an important feature for the roles. Also, the dashboard should have the ability to set policies for the roles. Something we could do in future. I don't remember if we have a tracker created for it.

Reply:
Alrighty then! Thanks! 
Yes we have https://tracker.ceph.com/issues/64351